server/platform/services/sharedchannel/channelinvite.go

```diff
@@ -172,10 +173,28 @@ func (scs *Service) onReceiveChannelInvite(msg model.RemoteClusterMsg, rc *model
 	}
 
 	// create new local channel to sync with the remote channel
-	if channel, err = scs.handleChannelCreation(invite, rc); err != nil {
+	if channel, created, err = scs.handleChannelCreation(invite, rc); err != nil {
 		return err
 	}
 
+	// ensure the channel returned was newly created by checking for the expected channel ID.
+	if invite.ChannelId != channel.Id {
```

Comment:
Doesn't `createDirectChannel` already handle this? https://github.com/mattermost/mattermost/pull/27357/files#diff-0fa8588e0aca051baee92532d5934eecce2df0995865b03543e6c6b0c5d3c4a3R341-R343

Also, the comment seems misleading since comparing IDs doesn't tell you whether the channel was newly created; that's what the `created` boolean is for.

Reply:
For direct channels that indeed does already check that the channel id is as expected, however public channels assumes a new channel is always created.  While that should normally hold true, I would like to keep this sanity check to detect any behaviour changes upstream.  If the ID's not match, sync won't work as expected and will fail silently.  I will update the comment.